homeassistant/components/blebox/cover.py

```diff
@@ -28,18 +36,39 @@
 }
 
 
+class BleboxCoverState(IntEnum):
+    """Internal cover state as reported by switchbox devices and blebox_uniapi."""
+
+    MOVING_DOWN = 0
+    MOVING_UP = 1
+    MANUALLY_STOPPED = 2
+    LOWER_LIMIT_REACHED = 3
+    UPPER_LIMIT_REACHED = 4
+    OVERLOAD = 5
+    MOTOR_FAILURE = 6
+    UNUSED = 7
+    SAFETY_STOP = 8
+
+
 BLEBOX_TO_HASS_COVER_STATES = {
     None: None,
-    0: STATE_CLOSING,  # moving down
-    1: STATE_OPENING,  # moving up
-    2: STATE_OPEN,  # manually stopped
-    3: STATE_CLOSED,  # lower limit
-    4: STATE_OPEN,  # upper limit / open
-    # gateController
-    5: STATE_OPEN,  # overload
-    6: STATE_OPEN,  # motor failure
-    # 7 is not used
-    8: STATE_OPEN,  # safety stop
+    # === switchBox states ===
+    BleboxCoverState.MOVING_DOWN: STATE_CLOSING,
+    BleboxCoverState.MOVING_UP: STATE_OPENING,
+    # note: MANUALLY_STOPPED means either "stopped during opening" or "stopped during
+    #       closing" but never fully closed or fully opened. However, CoverEntity does
+    #       not distinguish between such states. This may result in seemingly strange
+    #       state transitions in the event log like "Is closing" -> "Was opened".
+    #       This is fine as something that is "partially" opened isn't really closed.
+    #       Such presentation of the state is thus safer for users relying on it.
+    BleboxCoverState.MANUALLY_STOPPED: STATE_OPEN,
+    BleboxCoverState.LOWER_LIMIT_REACHED: STATE_CLOSED,
+    BleboxCoverState.UPPER_LIMIT_REACHED: STATE_OPEN,
+    # === switchBox + gateController states ===
+    BleboxCoverState.OVERLOAD: STATE_PROBLEM,
+    BleboxCoverState.MOTOR_FAILURE: STATE_PROBLEM,
```

Comment:
I'm aware of that. As you have also noticed in other comment the same is for jammed state (which is closest conceptually to safety stop as it means engine has stopped because gate/shutter driver discovered obstacle).

It's a pity that neither STATE_JAMMED nor STATE_PROBLEM is in any way observed by HA covers. On the other hand, existence of this state here has no negative impact on integration. 

What if we leave it here as is and let me figure out if I can make the base cover somehow observe problem/jammed state. This will be a bit more challenging to me as I need to look how such states are handled/implemented in other components. I could then submit such change as a separate PR so other integrations could benefit from it?

Reply:
I see. The end result is that the cover will in states problem and jammed get the "open" state, right?